packages/chain-bridge/src/router/BridgeRouter.sol

```diff
@@ -348,48 +458,129 @@ contract BridgeRouter is
+        // Validate adapter supports fleet deposits specifically
+        address specifiedAdapter = params.options.specifiedAdapter;
+        if (specifiedAdapter == address(0)) revert NoSuitableAdapter();
+        if (!this.isValidAdapter(specifiedAdapter)) revert UnknownAdapter();
+
+        _validateAdapterSupportsOperation(
+            specifiedAdapter,
+            BridgeTypes.OperationType.TRANSFER_ASSET
         );
 
-        // Apply fee buffer to account for fee volatility
-        uint256 bufferedFee = _applyFeeBuffer(requiredBaseFee);
+        // Get fee quote for fleet deposit operation
+        (uint256 requiredBaseFee, ) = IBridgeAdapter(specifiedAdapter)
+            .estimateFee(
+                params.destinationChainId,
+                params.asset,
+                params.amount,
+                params.options.adapterParams,
+                BridgeTypes.OperationType.TRANSFER_ASSET
+            );
 
-        // Validate fee provided by BridgeQueue against buffered fee
+        // Apply fee buffer and validate
+        uint256 bufferedFee = _applyFeeBuffer(requiredBaseFee);
         _validateFee(msg.value, bufferedFee);
 
-        _validateAdapterSupportsOperation(
-            specifiedAdapter,
-            BridgeTypes.OperationType.READ_STATE
+        // Transfer tokens from caller to adapter via router
+        _transferTokensToAdapter(
+            params.asset,
+            params.amount,
+            msg.sender,
+            specifiedAdapter
         );
 
-        // Generate the operation ID ONCE - Router is the source of truth
-        operationId = _generateOperationId(
+        // Generate operation ID for fleet deposit
+        operationId = _generateAndSetupOperation(
+            BridgeTypes.OperationType.TRANSFER_ASSET,
+            params.destinationChainId,
+            params.asset,
+            params.amount,
+            params.shareRecipient,
+            abi.encode(
+                params.fleetCommander,
+                params.shareRecipient,
+                params.originalUser
+            ),
+            specifiedAdapter
+        );
+
+        // Call adapter's transferAsset method instead of fleet deposit specific method
+        ISendAdapter(specifiedAdapter).transferAsset{value: bufferedFee}(
+            operationId,
+            params.destinationChainId,
+            params.asset,
+            params.shareRecipient,
+            params.amount,
+            params.originalUser,
+            params.originalUser, // Refund address is the original user
+            params.message,
+            params.options.adapterParams
+        );
+
+        // Emit fleet deposit specific event
+        emit FleetDepositInitiated(
+            operationId,
+            params.destinationChainId,
+            params.asset,
+            params.amount,
+            params.fleetCommander,
+            params.shareRecipient,
+            specifiedAdapter
+        );
+
+        return operationId;
+    }
```

Comment:
🛑 This method is transferring the asset tot the user on destination chain - there is no handling for the `USER_FLEET_DEPOSIT_TYPE`. 

The tests only check that the adapter is called and that the event is emitted, but do not verify that the fleetCommander is actually used as the recipient.